plugins/outputs/wavefront/sample.conf

```diff
@@ -1,19 +1,43 @@
 [[outputs.wavefront]]
-  ## Url for Wavefront API or Wavefront Proxy instance.
+  ## Url for Wavefront API or Wavefront proxy instance.
   url = "https://metrics.wavefront.com"
 
-  ## Authentication Token for Wavefront. Required if using Direct Ingestion. Not required if using a Wavefront Proxy.
-  #token = "DUMMY_TOKEN"
-
-  ## Maximum number of metrics to send per HTTP request. This value should be higher than the `metric_batch_size`. Default is 10,000. Values higher than 40,000 are not recommended.
-  # http_maximum_batch_size = 10000
-
-  ## Deprecated. DNS name of the Wavefront server or Wavefront Proxy. Use the `url` field instead.
+  ## Deprecated. DNS name of the Wavefront server or Wavefront proxy.
+  ## Use the `url` field instead.
+  ## The `host` and `port` below are equivalent to `url = "http://wavefront.example.com:2878"`
   #host = "wavefront.example.com"
 
   ## Deprecated. Wavefront proxy port. Use the `url` field instead.
   #port = 2878
 
+  ## Authentication for Direct Ingestion.
+  ## Direct Ingestion requires one of: `token`,`auth_csp_api_token`, or `auth_csp_client_credentials`
+  ## See https://docs.wavefront.com/csp_getting_started.html to learn more about using CSP credentials with Wavefront.
+  ## Not required if using a Wavefront proxy.
+
+  ## Wavefront API Token Authentication. Ignored if using a Wavefront proxy.
+  # 1. Click the gear icon at the top right in the Wavefront UI.
+  # 2. Click your account name (usually your email)
+  # 3. Click *API access*.
+  #token = "YOUR_TOKEN"
+
+  ## CSP API Token Authentication for Wavefront. Ignored if using a Wavefront proxy.
+  #auth_csp_api_token=CSP_API_TOKEN_HERE
+
+  ## CSP Client Credentials Authentication Information for Wavefront. Ignored if using a Wavefront proxy.
+  ## See also: https://docs.wavefront.com/csp_getting_started.html#whats-a-server-to-server-app
+  # [outputs.wavefront.auth_csp_client_credentials]
```

Comment:
This will not work due to TOML specifics. If you add the table here **all following options**, including `auth_csp_base_url`, `http_maximum_batch_size` etc, will be a child of the `outputs.wavefront.auth_csp_client_credentials` table. Therefore, those tables must be **at the end** of the config section as a table can only be ended by an array (`[[...]]`) or another table (`[...]`)...

Reply:
Thanks! We'll fix this.